### PR TITLE
Camel 13402 - Update to Californium 2.0.x and support DTLS

### PR DESCRIPTION
**Pull request by coheigea**
This task is to update to Californium 2.0.x and support DTLS in the camel-coap component. Currently we only support plain UDP and not UDP / TLS.

### CODE REVIEW

File: components/camel-coap/src/main/resources/META-INF/services/org/apache/camel/component/coaps [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#
```
Comment (oscerd) [inline components/camel-coap/src/main/resources/META-INF/services/org/apache/camel/component/coaps]: on master the META-INF is generated automatically during the component build, so you can safely remove this

Comment (coheigea) [reply components/camel-coap/src/main/resources/META-INF/services/org/apache/camel/component/coaps]: If I remove it then the build fails (mvn clean install in components/camel-coap)?

Comment (oscerd) [reply components/camel-coap/src/main/resources/META-INF/services/org/apache/camel/component/coaps]: you need to add the coaps scheme here:

https://github.com/apache/camel/pull/2891/files#diff-aa2b51fc80674de4afee8a0e0374a3e3R47

So it needs to look like:

@Component("coap,coaps")

Comment (coheigea) [reply components/camel-coap/src/main/resources/META-INF/services/org/apache/camel/component/coaps]: Thanks, fixed.

**Comment by coheigea** [comment]
@oscerd No idea on the camel-coap-starter question.

**Comment by oscerd** [comment]
I merged the PR on master. The Karaf feature didn't work, the californium-osgi bundle seems to be broken, so I had to wrap the normal JARs. Merged btw. Thanks

**Pull request closed without merging**